src/tools/miri/tests/native-lib/pass/ptr_read_access.rs

```diff
@@ -0,0 +1,89 @@
+//@only-target-linux
+//@only-on-host
+
+fn main() {
+    test_pointer();
+
+    test_simple();
+
+    test_nested();
+
+    test_static();
+}
+
+// Test function that dereferences a pointer and prints its contents from C.
+fn test_pointer() {
+    extern "C" {
+        fn print_pointer(ptr: *mut i32);
+    }
+
+    let mut x = 42;
+    let ptr = &mut x as *mut i32;
+
+    unsafe { print_pointer(ptr) };
+}
+
+// Test function that dereferences a simple struct pointer and accesses a field.
+fn test_simple() {
+    #[repr(C)]
+    struct Simple {
+        field: i32
+    }
+
+    extern "C" {
+        fn access_simple(s_ptr: *mut Simple) -> i32;
+    }
+
+    let mut simple = Simple { field: -42 };
+    let s_ptr = &mut simple as *mut Simple;
+
+    let result = unsafe { access_simple(s_ptr) };
+    assert_eq!(result, -42);
+}
+
+// Test function that dereferences nested struct pointers and accesses fields.
+fn test_nested() {
+    use std::ptr::NonNull;
+    
+    #[derive(Debug, PartialEq, Eq)]
+    #[repr(C)]
+    struct Nested {
+        value: i32,
+        next: Option<NonNull<Nested>>,
+    }
+
+    extern "C" {
+        fn access_nested(n_ptr: *mut Nested) -> i32;
+    }
+
+    let mut nested_0 = Nested { value: 0, next: None };
```

Comment:
Maybe put a more specific value than `0` here?

Reply:
Yeah that probably makes more sense - the values in this test are now (100, 99, 98, )97.